goaway.go

```diff
@@ -148,10 +179,13 @@ func (g *ProfanityDetector) Censor(s string) string {
 		currentIndex := 0
 		for currentIndex != -1 {
 			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
-				for i := 0; i < len(word); i++ {
-					censored = censored[:originalIndexes[foundIndex+currentIndex+i]] + "*" + censored[originalIndexes[foundIndex+currentIndex+i]+1:]
+				for i := 0; i < len([]rune(word)); i++ {
+					runeIndex := g.indexToRune(censored, currentIndex+foundIndex+i)
+					prefix := g.takeRunesToIndex(censored, originalIndexes[runeIndex])
+					postfix := g.takeRunesFromIndex(censored, originalIndexes[runeIndex]+1)
```

Comment:
Is there any chance these three could be merged into one?
I haven't run a benchmark, but I feel this might add a noticeable overhead 🤔

Reply:
Actually, scratch that. 
We may want to look into https://pkg.go.dev/unicode/utf8#DecodeRune instead:
> DecodeRune unpacks the first UTF-8 encoding in p and returns the rune and its width in bytes. If p is empty it returns (RuneError, 0). Otherwise, if the encoding is invalid, it returns (RuneError, 1). Both are impossible results for correct, non-empty UTF-8.

Reply:
Awesome!
Could you remove the `takeRunesFromIndex` and `takeRunesToIndex` functions then? 🙏

Reply:
yep, sorry, I missed those.